editors/sc-ide/core/main.cpp

```diff
@@ -128,72 +128,56 @@ int main( int argc, char *argv[] )
     return app.exec();
 }
 
+void SingleInstanceGuard::onNewIpcConnection()
+{
+	mIpcSocket = mIpcServer->nextPendingConnection();
+	connect(mIpcSocket, SIGNAL(disconnected()), mIpcSocket, SLOT(deleteLater()));
+	connect(mIpcSocket, SIGNAL(readyRead()), this, SLOT(onIpcData()));
+}
+
+void SingleInstanceGuard::onIpcLog(const QString &message) {
+	 
+}
 
 bool SingleInstanceGuard::tryConnect(QStringList const & arguments)
 {
-    const int maxNumberOfInstances = 128;
     if (!arguments.empty()) {
-        for (int socketID = 0; socketID != maxNumberOfInstances; ++socketID) {
-            QString serverName = QString("SuperColliderIDE_Singleton_%1").arg(socketID);
-            QSharedPointer<QLocalSocket> socket (new QLocalSocket(this));
-            socket->connectToServer(serverName);
-
-            QStringList canonicalArguments;
-            foreach (QString path, arguments) {
-                QFileInfo info(path);
-                canonicalArguments << info.canonicalFilePath();
-            }
-
-            if (socket->waitForConnected(200)) {
-                QDataStream stream(socket.data());
-                stream.setVersion(QDataStream::Qt_4_6);
-
-                stream << QString("open");
-                stream << canonicalArguments;
-                if (!socket->waitForBytesWritten(300))
-                    qWarning("SingleInstanceGuard: writing data to another IDE instance timed out");
-
-                return true;
-            }
+        QTcpSocket *socket = new QTcpSocket(this);
+		socket->connectToHost(QHostAddress(QHostAddress::LocalHost), SingleInstanceGuard::Port);
+		
+        QVariantList canonicalArguments;
+        foreach (QString path, arguments) {
+            QFileInfo info(path);
+            canonicalArguments << info.canonicalFilePath();
         }
-    }
 
-    mIpcServer = new QLocalServer(this);
-    for (int socketID = 0; socketID != maxNumberOfInstances; ++socketID) {
-        QString serverName = QString("SuperColliderIDE_Singleton_%1").arg(socketID);
+        if (socket->waitForConnected(200)) {
+			ScIpcChannel *ipcChannel = new ScIpcChannel(socket, QString("SingleInstanceGuard"), this);
+			ipcChannel->write(QString("open"), canonicalArguments);
 
-        bool listening = mIpcServer->listen(serverName);
-        if (listening) {
-            connect(mIpcServer, SIGNAL(newConnection()), this, SLOT(onNewIpcConnection()));
-            return false;
+			return true;
         }
     }
-    return false;
+
+    mIpcServer = new QTcpServer(this);
+	bool listening = mIpcServer->listen(QHostAddress(QHostAddress::LocalHost), SingleInstanceGuard::Port);
```

Comment:
i'm not adding any comments about mixing tabs and spaces anymore. please review all the code in an ide which can visualise them